django/test/client.py

```diff
@@ -174,9 +184,15 @@ def __call__(self, environ):
 
         # Emulate a WSGI server by calling the close method on completion.
         if response.streaming:
-            response.streaming_content = closing_iterator_wrapper(
-                response.streaming_content, response.close
-            )
+            if response.is_async:
+                response.streaming_content = aclosing_iterator_wrapper(
+                    response.streaming_content,
+                    response.close,
```

Comment:
I'd keep the same format in both calls
```suggestion
                    response.streaming_content, response.close
```